tests/test_spidermiddleware.py

```diff
@@ -328,12 +328,13 @@ def process_start_requests(self, start_requests, spider):
 class ProcessStartRequestsSimple(BaseAsyncSpiderMiddlewareTestCase):
     """process_start_requests tests for simple start_requests"""
 
-    ITEM_TYPE = Request
+    ITEM_TYPE = Union[Request, dict]
```

Comment:
Looks like this doesn't work on 3.9 because `isinstance()` can't use it.

Reply:
Changing this to tuple `ITEM_TYPE = (Request, dict)` resolve issues with `isinstance()` but it cause failing of `checks (3.8 typing-tests)`

Reply:
I doubt that, `isinstance()` supporting a tuple is unique to it, you cannot use a tuple instead of a union.

Reply:
(I haven't looked at this code yet but I suspect we need to rewrite it, e.g. by rewriting the assert)

Reply:
> I doubt that, isinstance() supporting a tuple is unique to it, you cannot use a tuple instead of a union.

It is directly mentioned on python docs that it support tuples as argument
https://docs.python.org/3.8/library/functions.html?highlight=tuple#isinstance
https://docs.python.org/3.9/library/functions.html?highlight=tuple#isinstance
https://docs.python.org/3.10/library/functions.html?highlight=tuple#isinstance
https://docs.python.org/3.11/library/functions.html?highlight=tuple#isinstance

built in `issubclass` - also supports tuple for its checks as well as `isinstance` so it is not unique.
As it is directly mentioned on docs - I think it's valid to use tuple here.

Reply:
I've meant that you can use a tuple for isinstance() but not for type hints. But I've now checked the actual failure, it's "Incompatible types in assignment (expression has type "Tuple[Type[Request], Type[Dict[Any, Any]]]", base class "BaseAsyncSpiderMiddlewareTestCase" defined the type as "type")" and it should be possible to solve it by updating the type hint of `BaseAsyncSpiderMiddlewareTestCase.ITEM_TYPE` to e.g. `Union[type, Union]` or something like that.